syntax = "proto3";

package protos;
option go_package = "orch-go/gen/protos;protos";
option csharp_namespace = "Core";

import "custom_types.proto";
import "google/protobuf/empty.proto";
import "notification.proto";

service NotificationAnalyticService {
  rpc ProcessGetAll (GetAllRequest) returns (GetAllNotificationsResponse);
  rpc ProcessGetById (GetNotificationByIdRequest) returns (NotificationModel);
  rpc ProcessGetByIds (GetNotificationByIdsRequest) returns (GetAllNotificationsResponse);
  rpc ProcessGetByDateRange (GetByDateRangeRequest) returns (GetAllNotificationsResponse);
  rpc ProcessGetByCustomer (GetNotificationsByCustomerRequest) returns (GetAllNotificationsResponse);
  rpc ProcessGetCount (google.protobuf.Empty) returns (CountResponse);
}
